diff -urN dnsmasq-2.93-orig/Makefile dnsmasq-2.93/Makefile
--- dnsmasq-2.93-orig/Makefile	2026-08-09 21:57:07.623705902 +0100
+++ dnsmasq-2.93/Makefile	2026-08-09 21:58:45.619621659 +0100
@@ -18,15 +18,12 @@
 
 # Variables you may well want to override.
 
-PREFIX        = /usr/local
+PREFIX        = /usr
 BINDIR        = $(PREFIX)/sbin
 MANDIR        = $(PREFIX)/share/man
 LOCALEDIR     = $(PREFIX)/share/locale
 BUILDDIR      = $(SRC)
 DESTDIR       = 
-CFLAGS        = -Wall -W -O2
-LDFLAGS       = 
-COPTS         = 
 RPM_OPT_FLAGS = 
 LIBS          = 
 LUA           = lua
diff -urN dnsmasq-2.93-orig/src/config.h dnsmasq-2.93/src/config.h
--- dnsmasq-2.93-orig/src/config.h	2026-08-09 21:57:07.630745904 +0100
+++ dnsmasq-2.93/src/config.h	2026-08-09 21:58:45.576132479 +0100
@@ -51,7 +51,7 @@
 #define DEFLEASE 3600 /* default DHCPv4 lease time, one hour */
 #define DEFLEASE6 (3600*24) /* default lease time for DHCPv6. One day. */
 #define CHUSER "nobody"
-#define CHGRP "dip"
+#define CHGRP "nogroup"
 #define TFTP_MAX_CONNECTIONS 50 /* max simultaneous connections */
 #define TFTP_MAX_WINDOW 32 /* max window size to negotiate */
 #define TFTP_TRANSFER_TIME 120 /* Abandon TFTP transfers after this long. Two mins. */
@@ -190,6 +190,7 @@
 #define HAVE_IPSET 
 #define HAVE_LOOP
 #define HAVE_DUMPFILE
+#define LEASEFILE "/tmp/dnsmasq.leases"
 
 /* Build options which require external libraries.
    
diff -urN dnsmasq-2.93-orig/src/dnsmasq.c dnsmasq-2.93/src/dnsmasq.c
--- dnsmasq-2.93-orig/src/dnsmasq.c	2026-08-09 21:57:07.630192797 +0100
+++ dnsmasq-2.93/src/dnsmasq.c	2026-08-09 21:58:45.580290043 +0100
@@ -1091,10 +1091,7 @@
   /* finished start-up - release original process */
   if (err_pipe[1] != -1)
     close(err_pipe[1]);
-  
-  if (daemon->port != 0)
-    check_servers(0);
-  
+
   pid = getpid();
 
 #ifdef HAVE_INOTIFY
@@ -1583,14 +1580,14 @@
 	  {
 	    if (daemon->resolv_files && option_bool(OPT_NO_POLL))
 	      {
-		reload_servers(daemon->resolv_files->name);
-		check = 1;
+		if (reload_servers(daemon->resolv_files->name))
+			check = 1;
 	      }
 
 	    if (daemon->servers_file)
 	      {
-		read_servers_file();
-		check = 1;
+		if (read_servers_file())
+			check = 1;
 	      }
 
 	    if (check)
diff -urN dnsmasq-2.93-orig/src/dnsmasq.h dnsmasq-2.93/src/dnsmasq.h
--- dnsmasq-2.93-orig/src/dnsmasq.h	2026-08-09 21:57:07.626861271 +0100
+++ dnsmasq-2.93/src/dnsmasq.h	2026-08-09 21:58:45.580406902 +0100
@@ -1564,7 +1564,7 @@
 char *option_string(int prot, unsigned int opt, unsigned char *val, 
 		    int opt_len, char *buf, int buf_len);
 void reread_dhcp(void);
-void read_servers_file(void);
+int read_servers_file(void);
 void set_option_bool(unsigned int opt);
 void reset_option_bool(unsigned int opt);
 struct hostsfile *expand_filelist(struct hostsfile *list);
diff -urN dnsmasq-2.93-orig/src/lease.c dnsmasq-2.93/src/lease.c
--- dnsmasq-2.93-orig/src/lease.c	2026-08-09 21:57:07.628517439 +0100
+++ dnsmasq-2.93/src/lease.c	2026-08-09 21:58:45.577618744 +0100
@@ -1064,9 +1064,10 @@
   struct dhcp_lease *lease_tmp;
   char *new_name = NULL, *new_fqdn = NULL;
 
+#if 0
   if (config_domain && (!domain || !hostname_isequal(domain, config_domain)))
     my_syslog(MS_DHCP | LOG_WARNING, _("Ignoring domain %s for DHCP host name %s"), config_domain, name);
-  
+#endif
   if (lease->hostname && name && hostname_isequal(lease->hostname, name))
     {
       if (auth)
diff -urN dnsmasq-2.93-orig/src/network.c dnsmasq-2.93/src/network.c
--- dnsmasq-2.93-orig/src/network.c	2026-08-09 21:57:07.628816954 +0100
+++ dnsmasq-2.93/src/network.c	2026-08-09 21:58:45.580175384 +0100
@@ -1604,7 +1604,9 @@
 	  break;
       if (iface)
 	{
+#if 0
 	  my_syslog(LOG_WARNING, _("ignoring nameserver %s - local interface"), daemon->namebuff);
+#endif
 	  serv->flags |= SERV_MARK;
 	  continue;
 	}
diff -urN dnsmasq-2.93-orig/src/option.c dnsmasq-2.93/src/option.c
--- dnsmasq-2.93-orig/src/option.c	2026-08-09 21:57:07.629867598 +0100
+++ dnsmasq-2.93/src/option.c	2026-08-09 21:58:45.580549007 +0100
@@ -5845,21 +5845,21 @@
   return list;
 }
 
-void read_servers_file(void)
+int read_servers_file(void)
 {
   FILE *f;
 
   if (!(f = fopen(daemon->servers_file, "r")))
     {
        my_syslog(LOG_ERR, _("cannot read %s: %s"), daemon->servers_file, strerror(errno));
-       return;
+       return 0;
     }
   
   mark_servers(SERV_FROM_FILE);
   read_file(daemon->servers_file, f, LOPT_REV_SERV, 0);
   fclose(f);
   cleanup_servers();
-  check_servers(0);
+  return 1;
 }
  
 
diff -urN dnsmasq-2.93-orig/src/rfc2131.c dnsmasq-2.93/src/rfc2131.c
--- dnsmasq-2.93-orig/src/rfc2131.c	2026-08-09 21:57:07.630350814 +0100
+++ dnsmasq-2.93/src/rfc2131.c	2026-08-09 21:58:45.579541747 +0100
@@ -1920,7 +1920,7 @@
 {
   char *mac = "";
 
-  if (!err && !option_bool(OPT_LOG_OPTS) && option_bool(OPT_QUIET_DHCP))
+  if (!option_bool(OPT_LOG_OPTS) && option_bool(OPT_QUIET_DHCP))
     return;
   
   daemon->addrbuff[0] = 0;
diff -urN dnsmasq-2.93-orig/src/rfc3315.c dnsmasq-2.93/src/rfc3315.c
--- dnsmasq-2.93-orig/src/rfc3315.c	2026-08-09 21:57:07.628140204 +0100
+++ dnsmasq-2.93/src/rfc3315.c	2026-08-09 21:58:45.578282696 +0100
@@ -189,8 +189,10 @@
 	  
       if (!state->context)
 	{
+#if 0
 	  my_syslog(MS_DHCP | LOG_WARNING, 
 		    _("no address range available for DHCPv6 request via %s"), state->iface_name);
+#endif
 	  return 0;
 	}
 
@@ -818,7 +820,7 @@
 	    for (c = state->context; c; c = c->current)
 	      if (!(c->flags & CONTEXT_RA_STATELESS))
 		{
-		  log6_packet(state, state->lease_allocate ? "DHCPREPLY" : "DHCPADVERTISE", NULL, _("no addresses available"));
+		  log6_quiet(state, state->lease_allocate ? "DHCPREPLY" : "DHCPADVERTISE", NULL, _("no addresses available"));
 		  break;
 		}
 	  }
@@ -934,7 +936,7 @@
 	    put_opt6_short(DHCP6NOADDRS);
 	    put_opt6_string(_("no addresses available"));
 	    end_opt6(o1);
-	    log6_packet(state, "DHCPREPLY", NULL, _("no addresses available"));
+	    log6_quiet(state, "DHCPREPLY", NULL, _("no addresses available"));
 	  }
 
 	tagif = add_options(state, 0);
@@ -1049,9 +1051,6 @@
 		    message = _("address invalid");
 		  } 
 
-		if (message && (message != state->hostname))
-		  log6_packet(state, "DHCPREPLY", &req_addr, message);	
-		else
 		  log6_quiet(state, "DHCPREPLY", &req_addr, message);
 	
 		o1 =  new_opt6(OPTION6_IAADDR);
