diff -urN miniupnpd-2.3.10-orig/Makefile.linux miniupnpd-2.3.10/Makefile.linux
--- miniupnpd-2.3.10-orig/Makefile.linux	2025-05-25 23:10:20.000000000 +0100
+++ miniupnpd-2.3.10/Makefile.linux	2026-08-09 21:30:18.732983179 +0100
@@ -80,6 +80,13 @@
 
 #LDLIBS += -lnfnetlink
 
+LDLIBS  = $(IPTABLESPATH)/libiptc/.libs/libip4tc.a
+ifeq ($(ENABLE_IPV6),1)
+LDLIBS += $(IPTABLESPATH)/libiptc/.libs/libip6tc.a
+endif
+CFLAGS += -I$(IPTABLESPATH)/include
+LDFLAGS += -L$(IPTABLESPATH)/libiptc/.libs
+
 EXECUTABLES = miniupnpd miniupnpdctl
 
 ifeq ($(TESTS), 1)
diff -urN miniupnpd-2.3.10-orig/configure miniupnpd-2.3.10/configure
--- miniupnpd-2.3.10-orig/configure	2025-05-25 23:10:20.000000000 +0100
+++ miniupnpd-2.3.10/configure	2026-08-09 21:30:18.732786110 +0100
@@ -242,6 +242,9 @@
 fi
 fi
 
+# N56U special case
+OS_NAME=RT-N56U
+
 ${RM} ${CONFIGFILE}
 
 echo "/* MiniUPnP Project" >> ${CONFIGFILE}
@@ -572,6 +575,17 @@
 		echo "#endif" >> ${CONFIGFILE}
 		FW=iptables
 		;;
+	RT-N56U)
+		OS_FAMILY=Linux
+		OS_VERSION=3.4.3.9
+		OS_URL=https://github.com/OpenPadavan/padavan
+		FW=iptables
+		LEASEFILE=1
+		VENDORCFG=1
+		HAVE_IP_MREQN=1
+		DISABLEPPPCONN=1
+		TESTS=0
+		;;
 	Darwin)
 		MAKEFILE=Makefile.macosx
 		MAJORVER=`echo $OS_VERSION | cut -d. -f1`
@@ -629,76 +643,9 @@
 		MAKEFILE=Makefile.linux
 		echo "#define USE_NETFILTER 1" >> ${CONFIGFILE}
 		echo "#define USE_IPTABLES  1" >> ${CONFIGFILE}
-		if [ "$PKG_CONFIG" ] ; then
-			if pkg_detect --atleast-version=1.4.3 libiptc; then
-				IPTABLES_143=1
-			elif pkg_detect libiptc; then
-				:
-			else
-				echo "Warning: no libiptc pkg-config found"
-			fi
-			if pkg_detect --atleast-version=1.0.2 libnetfilter_conntrack \
-			              --atleast-version=1.0.3 libmnl; then
-				echo "CPPFLAGS += -DUSE_NFCT" >> ${CONFIG_MK}
-			fi
-		elif [ "$IPTABLESPATH" ] ; then
-			echo "CPPFLAGS += -I${IPTABLESPATH}/include/" >> ${CONFIG_MK}
-			echo "LDFLAGS += -L${IPTABLESPATH}/libiptc/" >> ${CONFIG_MK}
-			if [ "$OS_NAME" != "OpenWrt" ]; then
-				IPTABLESVERSION=`grep "\#define VERSION" ${IPTABLESPATH}/config.h | tr -d \" |cut -d" " -f3`
-				echo "detected libiptc version $IPTABLESVERSION"
-				echo "# detected libiptc version $IPTABLESVERSION" >> ${CONFIG_MK}
-				IPTVER1=`echo $IPTABLESVERSION | cut -d. -f1`
-				IPTVER2=`echo $IPTABLESVERSION | cut -d. -f2`
-				IPTVER3=`echo $IPTABLESVERSION | cut -d. -f3`
-				if [ $IPTVER1 -gt 1 ] || \
-				   [ \( $IPTVER1 -eq 1 \) -a \( \( $IPTVER2 -gt 4 \) \
-				     -o \( \( $IPTVER2 -eq 4 \) -a \( $IPTVER3 -ge 3 \) \) \) ] ; then
-					IPTABLES_143=1
-				fi
-				if [ "$IPTABLES_143" = "1" ] ; then
-					echo "LDLIBS += ${IPTABLESPATH}/libiptc/.libs/libip4tc.o" >> ${CONFIG_MK}
-				else
-					echo "LDLIBS += ${IPTABLESPATH}/libiptc/libiptc.a" >> ${CONFIG_MK}
-				fi
-			else
-				# OpenWrt
-				# check for system-wide iptables files. Test if iptables version >= 1.4.3
-				# the following test has to be verified :
-				if test -f /usr/include/iptables/internal.h && \
-				   grep -q "\#define IPTABLES_VERSION" /usr/include/iptables/internal.h ; then
-					IPTABLES_143=1
-					echo "LDLIBS += -liptc" >> ${CONFIG_MK}
-				fi
-				arch=`echo $OS_MACHINE | grep -q x86_64 && echo 64`
-				if test -f /usr/lib${arch}/libiptc.a ; then
-					echo "LDLIBS += -liptc /usr/lib${arch}/libiptc.a" >> ${CONFIG_MK}
-				fi
-			fi
-		else
-			# IPTABLESPATH not defined and no pkg-config
-			echo "WARNING: no pkg-config and IPTABLESPATH not defined. checking /usr/include/xtables.h"
-			# since 1.4.16, XTABLES_VERSION_CODE has been moved to xtables-version.h
-			if test -f /usr/include/xtables-version.h || \
-			   test -f /usr/include/xtables.h && \
-			   grep -q "XTABLES_VERSION_CODE" /usr/include/xtables.h ; then
-				IPTABLES_143=1
-				echo "LDLIBS += -liptc" >> ${CONFIG_MK}
-				if test -f /lib/libip4tc.so ; then
-					echo "LDLIBS += -lip4tc" >> ${CONFIG_MK}
-				fi
-				if test -f /lib/libip6tc.so ; then
-					echo "LDLIBS += -lip6tc" >> ${CONFIG_MK}
-				fi
-			fi
-		fi
 		echo "/* when IPTABLES_143 is defined, miniupnpd uses the new API" >> ${CONFIGFILE}
 		echo " * from libiptc 1.4.3+ */ " >> ${CONFIGFILE}
-		if [ "$IPTABLES_143" = "1" ] ; then
-			echo "#define IPTABLES_143" >> ${CONFIGFILE}
-		else
-			echo "#undef IPTABLES_143" >> ${CONFIGFILE}
-		fi
+		echo "#define IPTABLES_143" >> ${CONFIGFILE}
 		;;
 	nftables)
 		if ! [ "$OS_FAMILY" = "Linux" ] ; then
@@ -758,7 +705,9 @@
 # libcap variants
 case $OS_FAMILY in
 	Linux)
-		if [ "$PKG_CONFIG" ] ; then
+		if [ "$OS_NAME" = "RT-N56U" ] ; then
+			:
+		elif [ "$PKG_CONFIG" ] ; then
 			if pkg_detect libcap-ng; then
 				echo "#define HAS_LIBCAP_NG" >> ${CONFIGFILE}
 			elif pkg_detect libcap; then
@@ -773,7 +722,9 @@
 # UUID API
 case $OS_FAMILY in
 	Linux)
-		if pkg_detect uuid; then
+		if [ "$OS_NAME" = "RT-N56U" ] ; then
+			:
+		elif pkg_detect uuid; then
 			echo "#define LIB_UUID" >> ${CONFIGFILE}
 		else
 			echo "Warning: uuid pkg-config not found"
